From: Phil Dibowitz <phil@ipom.com>
Date: Sun, 31 Dec 2006 08:19:50 +0000 (-0800)
Subject: USB Storage: unusual_devs: add supertop drives
X-Git-Tag: v2.6.20-rc4~4^2~6
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=f1cd4ad202ce34d29c847fd82df948ff67c17826

USB Storage: unusual_devs: add supertop drives

This combines patches from Alan Stern and Robert Schedel for two "Super Top"
drives that need the IGNORE_RESIDUE flag but have different vendor IDs.

Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

Adjusted to apply to Debian's 2.6.18 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.18.orig/drivers/usb/storage/unusual_devs.h linux-source-2.6.18/drivers/usb/storage/unusual_devs.h
--- linux-source-2.6.18.orig/drivers/usb/storage/unusual_devs.h	2007-12-18 08:21:23.000000000 -0700
+++ linux-source-2.6.18/drivers/usb/storage/unusual_devs.h	2007-12-21 11:31:09.000000000 -0700
@@ -1287,6 +1287,21 @@ UNUSUAL_DEV(  0x1370, 0x6828, 0x0110, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_IGNORE_RESIDUE ),
 
+/* Reported by Francesco Foresti <frafore@tiscali.it> */
+UNUSUAL_DEV(  0x14cd, 0x6600, 0x0201, 0x0201,
+		"Super Top",
+		"IDE DEVICE",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
+/* Reported by Robert Schedel <r.schedel@yahoo.de>
+ * Note: this is a 'super top' device like the above 14cd/6600 device */
+UNUSUAL_DEV(  0x1652, 0x6600, 0x0201, 0x0201,
+		"Teac",
+		"HD-35PUK-B",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
 /* patch submitted by Davide Perini <perini.davide@dpsoftware.org>
  * and Renato Perini <rperini@email.it>
  */
